syntax = "proto3";

option go_package = "protogen/bike";

import "google/protobuf/empty.proto";

service Bike {
  rpc GetStationList(google.protobuf.Empty) returns (StationList) {}
  rpc GetRealTimeStationStatus(google.protobuf.Empty) returns (stream StationStatus) {}
}

// 응답으로만 쓰일 메시지 타입이므로 optional 키워드 제외함
message Station {
  string stn_grp_name = 1;
  string stn_id = 2;
  string stn_num = 3;
  string stn_name = 4;
  string stn_addr_1 = 5;
  string stn_addr_2 = 6;
  float stn_lat = 7;
  float stn_lng = 8;
  int32 hold_num = 9;
}

message StationList {
  int32 count = 1;
  repeated Station stations = 2;
}

message StationStatus {
  string stn_id = 1;
  string stn_name = 2;
  int32 parked_bike_cnt = 3;
}
